EIPS/eip-7783.md

```diff
@@ -0,0 +1,99 @@
+
+## **Motivation**
+
+### **Predictable Gas Limit Growth**
+
+- **Current Issue:**
+  - The Ethereum network faces increasing demand, but changes to the gas limit are often manually adjusted by miners or validators based on their preferences, which may cause unpredictable block sizes and network performance issues.
+
+- **Need for Change:**
+  - A systematic and predictable increase of the gas limit will help scale the network while giving the ecosystem time to adjust to larger block sizes, without needing to rely on ad hoc decisions by network participants.
+
+### **Gradual Increase with Deactivation Safeguard**
+
+- **Controlled Growth:**
+  - Instead of sudden or unpredictable changes, this EIP proposes incremental gas Limit increases over a specified amount of time, ensuring a smooth transition to higher transaction throughput, while still keeping the governance of the gas Limit in the hand of the community.
+  
+- **Automatic Deactivation:**
+  - A safeguard deactivation block will halt the increase after some specified time, preventing the gas Limit from growing indefinitely and allowing the community to reassess the network's needs.
+
+## **Specification**
+
+### **Incremental Gas Limit Increase Strategy**
+
+Add a new "Gas Limit" selection strategy that takes in Block Number `N` and spits out the Gas Limit `GL` for that block. The strategy is as follows:
+
+ - The gas limit `GL_t` at block `t` is calculated as:
+
+```python
+def compute_gas_limit(blockNum: int, blockNumStart: int, initialGasLimit: int, r: int, gasLimitCap: int) -> int:
+  if blockNum < blockNumStart:
+    return initialGasLimit
+  else:
+    return min(gasLimitCap, initialGasLimit + r * (blockNum - blockNumStart))
+```
+
+Where:
+
+- `blockNum` is the block number for which the gas limit is being calculated.
+- `blockNumStart` is the block number at which the gas limit increase starts.
+- `initialGasLimit` is the initial gas limit at block `blockNumStart`.
+- `r` is the rate at which the gas limit increases per block.
+- `gasLimitCap` is the maximum gas limit that can be reached.
+
+If we set `blockNumStart` to the current block number, `initialGasLimit` to the current gas limit (`30_000_000`), `r` to 6, and `gasLimitCap` to `60_000_000`, the gas limit will increase by 6 gas per block for two years, reaching 30 million gas at the end of a 2 years period. 
+
+### **Implementation**
+
+- **Client Updates:**
+  - Ethereum clients must implement the automatic gas limit increase according to the specified rate and formula, with enforcement of the deactivation block. However, If the GasLimit has been locally set by the validator that will override the formula and take precedence.
```

Comment:
```suggestion
  - Ethereum clients must implement the automatic gas limit increase according to the specified rate and formula, with enforcement of the deactivation block. However, if the gas limit has been locally set by the validator that will override the formula and take precedence.
```

I also find the second sentence "However..." formatted a bit weird :sweat_smile: